src/cmd/manage_nsfs.js

```diff
@@ -238,24 +260,42 @@ async function get_bucket_config_file_status(data, bucket_config_path, config_ro
         return;
     }
     try {
-        const config_data = await get_config_data(bucket_config_path, data.name);
+        const bucket_path = get_config_file_path(bucket_config_path, data.name);
+        const config_data = await get_view_config_data(bucket_path);
         console.log(config_data);
     } catch (err) {
-        console.log('Bucket do not exists with name: ' + data.name);
+        console.log('Bucket does not exist with name: ' + data.name);
     }
 }
 
 async function update_bucket_config_file(data, bucket_config_path, config_root_backend) {
-    const is_valid = await validate_bucket_add_args(data);
+    const is_valid = await validate_bucket_add_args(data, true);
     if (!is_valid) {
         print_bucket_usage();
         return;
     }
     // TODO: support non root fs context
     const fs_context = get_root_fs_context(config_root_backend);
-    const full_bucket_config_path = get_config_file_path(bucket_config_path, data.name);
-    data = JSON.stringify(data);
-    await native_fs_utils.update_config_file(fs_context, bucket_config_path, full_bucket_config_path, data);
+
+    const cur_name = data.name;
+    const update_name = data.new_name && cur_name && data.new_name.unwrap() !== cur_name.unwrap();
+
+    if (!update_name) {
+        const full_bucket_config_path = get_config_file_path(bucket_config_path, data.name);
+        data = JSON.stringify(data);
+        await native_fs_utils.update_config_file(fs_context, bucket_config_path, full_bucket_config_path, data);
+        return;
+    }
+
+    data.name = data.new_name || cur_name;
+
+    const cur_bucket_config_path = get_config_file_path(bucket_config_path, cur_name.unwrap());
+    const new_bucket_config_path = get_config_file_path(bucket_config_path, data.name.unwrap());
+
+    data = JSON.stringify(_.omit(data, ['new_name']));
+
+    await native_fs_utils.create_config_file(fs_context, bucket_config_path, new_bucket_config_path, data);
+    await native_fs_utils.delete_config_file(fs_context, bucket_config_path, cur_bucket_config_path);
```

Comment:
I think we should delete the previous file only when creation is success.

Reply:
what do you mean on success? currently it'll throw error if the creation will fail so it won't delete the old path , or I missed something?

Reply:
I missed that, ya make sense